packages/browsers/src/browser-data/firefox.ts

```diff
@@ -421,6 +422,27 @@ async function writePreferences(options: ProfileOptions): Promise<void> {
     const prefsBackupPath = path.join(options.path, 'prefs.js.puppeteer');
     await fs.promises.copyFile(prefsPath, prefsBackupPath);
   }
+
+  if (options.disableExtraUserContexts) {
+    const containersPath = path.join(options.path, 'containers.json');
+
+    const singleContainer = JSON.stringify({
```

Comment:
@whimboo we want to disable extra user contexts for Firefox in Puppeteer (for profiles created by Puppeteer). This works but I wonder if this solution is something we can rely on?

P.S. @Lightning00Blade did you notice any speed up in launching Firefox?

Reply:
With parallelization it seems the overhead of writing a file to be around `30-50ms` and that varies from run to run a lot have seen the old one be slower and the new one be faster.

Reply:
(whimboo is off this week)

There are two internal containers that should be preserved:
https://searchfox.org/mozilla-central/rev/6121b33709dd80979a6806ff59096a561e348ae8/toolkit/components/contextualidentity/ContextualIdentityService.sys.mjs#85-105

userContextIdInternal.thumbnail and userContextIdInternal.webextStorageLocal
I am not sure what are the side effects of removing the second one, but I think it's better to keep it.

Note that the current version was recently bumped to 5 (https://bugzilla.mozilla.org/show_bug.cgi?id=1814969), but it's fine to keep 4 for now, and anyway the contextualidentityservice has migration logic based on the version.

Overall I would say this is fine as a temporary solution but we should try to introduce a preference in Firefox in order to skip the default user profiles.

Reply:
@Lightning00Blade let's probably rewrite the tests to expect any number of non-default contexts.

Reply:
Yeah sound good, I think I would extract some of the changes from this PR as well, I think we can make FF launch just a little bit faster.